src/main/java/com/hivemq/cli/commands/AbstractConnectRestrictionFlags.java

```diff
@@ -13,144 +13,150 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.hivemq.cli.commands;
 
 import com.hivemq.client.mqtt.MqttVersion;
 import com.hivemq.client.mqtt.mqtt5.message.connect.Mqtt5ConnectRestrictions;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import org.tinylog.Logger;
 import picocli.CommandLine;
 
 public abstract class AbstractConnectRestrictionFlags extends AbstractWillFlags implements ConnectRestrictions {
 
-    @CommandLine.Option(names = {"--rcvMax"}, description = "The maximum amount of not acknowledged publishes with QoS 1 or 2 the client accepts from the server concurrently. (default: " + Mqtt5ConnectRestrictions.DEFAULT_RECEIVE_MAXIMUM + ")", order = 3)
-    @Nullable
-    private Integer receiveMaximum;
-
-    @CommandLine.Option(names = {"--sendMax"}, description = "The maximum amount of not acknowledged publishes with QoS 1 or 2 the client send to the server concurrently. (default: " + Mqtt5ConnectRestrictions.DEFAULT_SEND_MAXIMUM + ")", order = 3)
-    @Nullable
-    private Integer sendMaximum;
-
-    @CommandLine.Option(names = {"--maxPacketSize"}, description = "The maximum packet size the client accepts from the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_MAXIMUM_PACKET_SIZE + ")", order = 3)
-    @Nullable
-    private Integer maximumPacketSize;
-
-    @CommandLine.Option(names = {"--sendMaxPacketSize"}, description = "The maximum packet size the client sends to the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_SEND_MAXIMUM_PACKET_SIZE + ")", order = 3)
-    @Nullable
-    private Integer sendMaximumPacketSize;
-
-    @CommandLine.Option(names = {"--topicAliasMax"}, description = "The maximum amount of topic aliases the client accepts from the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_TOPIC_ALIAS_MAXIMUM + ")", order = 3)
-    @Nullable
-    private Integer topicAliasMaximum;
-
-    @CommandLine.Option(names = {"--sendTopicAliasMax"}, description = "The maximum amount of topic aliases the client sends to the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_SEND_TOPIC_ALIAS_MAXIMUM + ")", order = 3)
-    @Nullable
-    private Integer sendTopicAliasMaximum;
-
-    @CommandLine.Option(names = {"--reqProblemInfo"}, negatable = true, description = "The client requests problem information from the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_REQUEST_PROBLEM_INFORMATION + ")", order = 3)
-    @Nullable
-    private Boolean requestProblemInformation;
-
-    @CommandLine.Option(names = {"--reqResponseInfo"}, negatable = true, description = "The client requests response information from the server. (default: " + Mqtt5ConnectRestrictions.DEFAULT_REQUEST_RESPONSE_INFORMATION + ")", order = 3)
-    @Nullable
-    private Boolean requestResponseInformation;
+    @SuppressWarnings("unused")
```

Comment:
Mybe just supress for class? 🤷🏿

Reply:
I think the explicit suppression is better suited as only the picocli command definitions are definitely unused. Everything else in the class should be handled differently. I think the tradeoff of these additional lines are justified.